projects/packages/my-jetpack/_inc/components/welcome-flow/ConnectionStep.tsx

```diff
@@ -16,23 +21,50 @@ type ConnectionStepProps = {
  *
  * @param {object} props - ConnectioStepProps
  * @param {Function} props.onActivateSite - Alias for handleRegisterSite
+ * @param {Function} props.onUpdateWelcomeFlowExperiment - Updating the welcomeFlowExperiment state
  * @param {boolean} props.isActivating - Alias for siteIsRegistering
  * @returns {object} The ConnectionStep component.
  */
-const ConnectionStep = ( { onActivateSite, isActivating }: ConnectionStepProps ) => {
+const ConnectionStep = ( {
+	onActivateSite,
+	onUpdateWelcomeFlowExperiment,
+	isActivating,
+}: ConnectionStepProps ) => {
 	const { recordEvent } = useAnalytics();
 	const { setNotice, resetNotice } = useContext( NoticeContext );
 
 	const activationButtonLabel = __( 'Activate Jetpack in one click', 'jetpack-my-jetpack' );
 
-	const onConnectSiteClick = useCallback( () => {
+	const onConnectSiteClick = useCallback( async () => {
 		recordEvent( 'jetpack_myjetpack_welcome_banner_connect_site_click' );
-		onActivateSite().then( () => {
-			recordEvent( 'jetpack_myjetpack_welcome_banner_connect_site_success' );
-			resetNotice();
-			setNotice( NOTICE_SITE_CONNECTED, resetNotice );
-		} );
-	}, [ onActivateSite, recordEvent, resetNotice, setNotice ] );
+		onUpdateWelcomeFlowExperiment( state => ( { ...state, isLoading: true } ) );
+		await onActivateSite();
+
+		recordEvent( 'jetpack_myjetpack_welcome_banner_connect_site_success' );
+
+		await sideloadTracks();
+
+		initializeExPlat();
+
+		const { variationName } = await loadExperimentAssignment(
```

Comment:
Good question 🤔

Reply:
In theory, the call can timeout and/or fail. If that happens, it's likely that it has to do with the user's internet, then other requests would fail as well, but the function would fail silently. We can add a `try/catch` to be on the safe side, though.

Reply:
but in that case user would be stuck in "Activating..." state, right? let's get that try/catch working I think